feature/experimental/policy/policy_vrf_selection/otg_tests/protocol_dscp_rules_for_vrf_selection_test/protocol_dscp_rules_for_vrf_selection_test.go

```diff
@@ -234,80 +239,111 @@ func configInterfaceDUT(i *oc.Interface, dutPort *attrs.Attributes) *oc.Interfac
+	ethHeader := flow.Packet().Add().Ethernet()
+	ethHeader.Src().SetValue(src.MAC)
+	outerIPHeader := flow.Packet().Add().Ipv4()
+	outerIPHeader.Src().SetValue(src.IPv4)
+	outerIPHeader.Dst().SetValue(dst.IPv4)
+	outerIPHeader.Priority().Dscp().Phb().SetValue(dscp)
+	innerIPHeader := flow.Packet().Add().Ipv4()
+	innerIPHeader.Src().SetValue("198.51.100.1")
+	innerIPHeader.Dst().Increment().SetStart("203.0.113.1").SetStep("0.0.0.1").SetCount(10000)
+
+	flow.Size().SetFixed(1024)
+	flow.Rate().SetPps(100)
+	flow.Duration().FixedPackets().SetPackets(100)
+
 	return flow
 }
 
 // testTrafficFlows verifies traffic for one or more flows.
-func testTrafficFlows(t *testing.T, args *testArgs, expectPass bool, flow ...*ondatra.Flow) {
+func testTrafficFlows(t *testing.T, args *testArgs, expectPass bool, flows ...gosnappi.Flow) {
+
+	args.top.Flows().Clear()
+	for _, flow := range flows {
+		args.top.Flows().Append(flow)
+	}
+	args.ate.OTG().PushConfig(t, args.top)
+	args.ate.OTG().StartProtocols(t)
+
+	t.Logf("*** Starting traffic ...")
+	args.ate.OTG().StartTraffic(t)
+	time.Sleep(trafficDuration)
+	t.Logf("*** Stop traffic ...")
+	args.ate.OTG().StopTraffic(t)
 
-	args.ate.Traffic().Start(t, flow...)
-	time.Sleep(60 * time.Second)
-	args.ate.Traffic().Stop(t)
 	if expectPass {
 		t.Log("Expecting traffic to pass for the flows")
 	} else {
 		t.Log("Expecting traffic to fail for the flows")
 	}
 
-	// log stats
-	t.Log("All flow LossPct: ", args.ate.Telemetry().FlowAny().LossPct().Get(t))
-	t.Log("FlowAny InPkts  : ", args.ate.Telemetry().FlowAny().Counters().InPkts().Get(t))
-	t.Log("FlowAny OutPkts : ", args.ate.Telemetry().FlowAny().Counters().OutPkts().Get(t))
+	top := args.ate.OTG().FetchConfig(t)
+	otgutils.LogFlowMetrics(t, args.ate.OTG(), top)
+	for _, flow := range flows {
+		t.Logf("*** Verifying %v traffic on OTG ... ", flow.Name())
+		outPkts := gnmi.Get(t, args.ate.OTG(), gnmi.OTG().Flow(flow.Name()).Counters().OutPkts().State())
+		inPkts := gnmi.Get(t, args.ate.OTG(), gnmi.OTG().Flow(flow.Name()).Counters().InPkts().State())
+		var lossPct uint64
+		if outPkts == 0 {
+			t.Logf("The loss percentage can't be calculated if no packets are sent.")
+		} else {
+			lossPct = ((outPkts - inPkts) * 100) / outPkts
+		}
 
-	flowPath := args.ate.Telemetry().FlowAny()
-	if got := flowPath.LossPct().Get(t); len(got) == 0 {
-		t.Fatalf("Flow stats count not correct, wanted > 0, got 0")
-	} else {
-		for i, lossPct := range got {
-			if (expectPass == true) && (lossPct == 0) {
-				t.Logf("Traffic for %v flow is passing as expected", flow[i].Name())
-			} else if (expectPass == false) && (lossPct == 100) {
-				t.Logf("Traffic for %v flow is failing as expected", flow[i].Name())
-
-			} else {
-				t.Fatalf("Traffic is not working as expected for flow: %v.", flow[i].Name())
-			}
+		// log stats
+		t.Log("All flow LossPct: ", lossPct)
+		t.Log("FlowAny InPkts  : ", inPkts)
+		t.Log("FlowAny OutPkts : ", outPkts)
```

Comment:
Can you update these logs to be singular, now that we're not using FlowAny and logging this for each flow individually?

e.g. t.Log("Flow InPkts  : ", inPkts) would work

Reply:
Done, modified